__tests__/app.test.js

```diff
@@ -157,41 +155,78 @@ describe("GET /api/articles/:article_id/comments", () => {
     return request(app)
-      .get('/api/articles/2/comments')
+      .get("/api/articles/2/comments")
       .expect(200)
       .then(({ body }) => {
         const { comments } = body;
-        expect(Array.isArray(comments)).toBe(true); 
-        expect(comments).toHaveLength(0)
-      })
-      })
-      it("400: returns bad request when invalid ID", () => {
-        return request(app)
-          .get("/api/articles/'apple'/comments")
-          .expect(400)
-          .then(({ body }) => {
-            expect(body.msg).toBe("Bad Request");
-          })
-      });
-      it("404: returns Not Found error when invalid ID does not exist", () => {
-        return request(app)
-          .get("/api/articles/9090/comments")
-          .expect(404)
-          .then(({ body }) => {
-            expect(body.msg).toBe("Not Found");
-          })
-      });
-  })
+        expect(Array.isArray(comments)).toBe(true);
+        expect(comments).toHaveLength(0);
+      });
+  });
+  it("400: returns bad request when invalid ID", () => {
+    return request(app)
+      .get("/api/articles/'apple'/comments")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad Request");
+      });
+  });
+  it("404: returns Not Found error when invalid ID does not exist", () => {
+    return request(app)
+      .get("/api/articles/9090/comments")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Not Found");
+      });
+  });
+});
 
+describe("POST /api/articles/:article_id/comments", () => {
+  it("201: should add a comment for an article", () => {
+    return request(app)
+      .post("/api/articles/5/comments")
+      .send({ username: "butter_bridge", body: "apple" })
+      .expect(201)
+      .then(({ body }) => {
+        const { comment } = body;
+
+        expect(comment).toHaveProperty("comment_id", expect.any(Number));
+        expect(comment).toHaveProperty("votes", expect.any(Number));
+        expect(comment).toHaveProperty("created_at", expect.any(String));
+        expect(comment).toHaveProperty("author", "butter_bridge");
+        // expect(comment).toHaveProperty('username', 'John');
+        expect(comment).toHaveProperty("body", "apple");
+        expect(comment).toHaveProperty("article_id", expect.any(Number));
+      });
+  });
+  it("404: should return Not Found if article does not exist", () => {
+    return request(app)
+      .post("/api/articles/999/comments")
+      .send({ username: "butter_bridge", body: "apple" })
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Not Found");
+      });
+  });
+  it("400: returns bad request when invalid ID", () => {
+    return request(app)
+      .post("/api/articles/'apple'/comments")
+      .send({ username: "butter_bridge", body: "apple" })
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad Request");
+      });
+  });
+});
```

Comment:
You are missing two tests here. 

404 - Username not found (you will _not_ need a check exists, psql will throw it's own error which can be caught with another if statement in your psql handler) 

400 - Missing keys, when you try to send {notusername: "butter_bridge", notbody: "valid text"} for example, again psql will throw an error for this, you shouldn't need promise.reject